dev/com.ibm.ws.security.oauth/test/com/ibm/oauth/core/test/base/BaseClient.java

```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.oauth.core.test.base;
+
+import com.google.gson.JsonArray;
+import com.ibm.oauth.core.api.oauth20.client.OAuth20Client;
+
+public class BaseClient implements OAuth20Client {
+
+    String _clientId;
+    String _clientSecret;
+    String _clientName;
+    JsonArray _redirectURIs;
+    boolean _isEnabled;
+    boolean _allowRegex;
+
+    public BaseClient(String clientId, String clientSecret, String clientName,
+                      JsonArray redirectURIs, boolean isEnabled) {
+        _clientId = clientId;
+        _clientSecret = clientSecret;
+        _clientName = clientName;
+        _redirectURIs = redirectURIs;
+        _isEnabled = isEnabled;
+    }
+
+    @Override
+    public String getClientId() {
+        return _clientId;
+    }
+
+    @Override
+    public String getClientSecret() {
+        return _clientSecret;
+    }
+
+    @Override
+    public String getClientName() {
+        return _clientName;
+    }
+
+    @Override
+    public JsonArray getRedirectUris() {
+        return _redirectURIs;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return _isEnabled;
+    }
+
+    @Override
+    public String[] getExtensionProperty(String propertyName) {
+        // No extension properties are expected
+        return null;
+    }
+
+    @Override
+    public String toString() {
+        StringBuffer sb = new StringBuffer();
+        sb.append("{");
+        sb.append("_clientId: " + _clientId);
+        sb.append(" _clientSecret: " + _clientSecret);
```

Comment:
Should we now skip printing the _clientSecret? Or leave it for for testing purposes?

Reply:
This class is only used for testing so printing the client secret isn't a problem. I'm skeptical about the need for this class to exist at all, but for now I was just bringing everything over as-is.